x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/Driver.java

```diff
@@ -253,6 +277,7 @@ private IsBlockedResult runSingleLoopIteration() {
             if (op.isBlocked().listener().isDone() == false) {
                 continue;
             }
+            checkForEarlyTermination();
```

Comment:
Where are these `checkForEarlyTermination` calls done (here and 2 more below)? Are they only done on the coordinator or on all data nodes running some part of the query?

Reply:
These happen for every pipeline, including the leaf data nodes.